### PR TITLE
PHPORM-211 Fix unsetting property in embedded model

### PR DESCRIPTION
**Pull request by GromNaN**
Fix PHPORM-211

### Checklist

- [x] Add tests and ensure they pass
- [x] Add an entry to the CHANGELOG.md file
- [ ] Update documentation for new features

### CODE REVIEW

File: src/Relations/EmbedsOneOrMany.php [hunk 1]
```diff
@@ -392,7 +393,12 @@ public static function getUpdateValues($array, $prepend = '')
         $results = [];
 
         foreach ($array as $key => $value) {
-            $results[$prepend . $key] = $value;
+            if ($key === '$unset') {
+                assert(is_array($value), '$unset must be an array.');
+                $results[$key] = static::getUpdateValues($value, $prepend);
+            } else {
+                $results[$prepend . $key] = $value;
+            }
```
Comment (GromNaN) [inline src/Relations/EmbedsOneOrMany.php:401]: Before, the incorrect update was:
```ts
{
  $set: {
    "address.$.updated_at": Date(...),
    "address.$.$unset": {
      "city": true,
    }
   }
}
```

After, it becomes:
```ts
{
  "$unset": {
    "address.$.city": true,
  },
  "$set": {
    "address.$.updated_at": Date(...),
  }
}
```

File: src/Relations/EmbedsOneOrMany.php [hunk 2]
```diff
@@ -392,7 +394,12 @@ public static function getUpdateValues($array, $prepend = '')
         $results = [];
 
         foreach ($array as $key => $value) {
-            $results[$prepend . $key] = $value;
+            if (str_starts_with($key, '$')) {
+                assert(is_array($value), 'Update operator value must be an array.');
+                $results[$key] = static::getUpdateValues($value, $prepend);
+            } else {
+                $results[$prepend . $key] = $value;
+            }
```
Comment (GromNaN) [inline src/Relations/EmbedsOneOrMany.php:402]: Before, the incorrect update was:
```ts
{
  $set: {
    "address.$.updated_at": Date(...),
    "address.$.$unset": {
      "city": true,
    }
   }
}
```

After, it becomes:
```ts
{
  "$unset": {
    "address.$.city": true,
  },
  "$set": {
    "address.$.updated_at": Date(...),
  }
}
```

Field names starting with a `$` are not supported, that's why we are sure there will not be 2 levels of recursion.

File: tests/EmbeddedRelationsTest.php [hunk 3]
```diff
@@ -25,14 +19,7 @@ class EmbeddedRelationsTest extends TestCase
...
     {
         Mockery::close();
-
         User::truncate();
-        Book::truncate();
-        Item::truncate();
-        Role::truncate();
-        Client::truncate();
-        Group::truncate();
-        Photo::truncate();
```
Comment (GromNaN) [inline tests/EmbeddedRelationsTest.php:35]: Side change: This models are not used in this test class.

File: tests/EmbeddedRelationsTest.php [hunk 4]
```diff
@@ -951,4 +938,36 @@ public function testGetQueueableRelationsEmbedsOne()
...
+        $this->assertSame(null, $user->addresses->get(0)->city);
+        $this->assertSame('Tokyo', $user->addresses->get(1)->city);
+
+        // Unset and reset property
+        unset($user->addresses->get(1)->city);
+        $user->addresses->get(1)->city = 'Kyoto';
+        $user->addresses->get(1)->save();
+
+        $user = User::where('name', 'John Doe')->first();
+        $this->assertSame(null, $user->addresses->get(0)->city);
```
Comment (alcaeus) [inline tests/EmbeddedRelationsTest.php]:
```suggestion
        $this->assertNull($user->addresses->get(0)->city);
```

File: tests/EmbeddedRelationsTest.php [hunk 5]
```diff
@@ -951,4 +938,36 @@ public function testGetQueueableRelationsEmbedsOne()
...
+        $user = User::where('name', 'John Doe')->first();
+        $this->assertSame('Paris', $user->addresses->get(0)->city);
+        $this->assertSame('Tokyo', $user->addresses->get(1)->city);
+
+        // Unset property
+        unset($user->addresses->first()->city);
+        $user->addresses->first()->save();
+
+        $user = User::where('name', 'John Doe')->first();
+        $this->assertSame(null, $user->addresses->get(0)->city);
```
Comment (alcaeus) [inline tests/EmbeddedRelationsTest.php]:
```suggestion
        $this->assertNull($user->addresses->get(0)->city);
```

**Review by alcaeus — COMMENTED** [review]
Nit about the assertion method, but LGTM other than that!

File: tests/EmbeddedRelationsTest.php (hunk 4 above)
Comment (GromNaN) [reply tests/EmbeddedRelationsTest.php]: I've done this to be consistent with the other asserts, but it's just a very personal perspective.

**Review by alcaeus — APPROVED** [review]

**Pull request merged**